capstone-project/src/main/angular-webapp/src/app/map/map.component.ts

```diff
@@ -34,6 +32,7 @@ export class MapComponent implements OnInit {
 
     // Editable marker that displays when a user clicks on the map.
     let editableMarker;
```

Comment:
Overall this looks much better. 

editableMarker should probably be a class member, and all functions should be defined as call methods rather than scoped on the ngOnInit method.

But that can be in a following PR, I know you worked hard on this one.

Reply:
Thanks Or! We will fix that in a separate PR